### PR TITLE
comments

### PR DESCRIPTION
**Pull request by clburns107**
pull request for comments

### CODE REVIEW

**Comment by sumeetjain** [comment]
> Should I write a test for every combination I can think of?

Yes, but start simple. Write _two_, then submit for feedback. You really don't need this to be perfect–it's an evolving process.

**Comment by clburns107** [comment]
are we to run the tests file or rake with it? I dont think either would work since I dont have a method yet but I'm not sure which one's errors I'm supposed to pay attention to.

**Comment by halfghaninne** [comment]
You will `rake` your program in the command line from the root folder. If your tests are written correctly, they should run and fail, since your method is not yet written. If the rake aborts, or you get some other error other than the tests running and failing. there may be an issue with the way the tests are written. 

As a reminder, you will also need to write methods for DictionaryTest.

**Comment by sumeetjain** [comment]
- [ ] Add some more tests that cover _simple_ scenarios too.
- [ ] Add tests for the dictionary file.

**Comment by clburns107** [comment]
ok so i had the string segmenter tests raking so i moved on to the dictionary tests. I am getting a syntax error back for line 10. Ive gone so far as to delete everything in the file. Im still receiving the same error and nothing will rake now.

**Comment by clburns107** [comment]
sorry here is the error I get.

/Users/mac/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/rake_test_loader.rb: /Users/mac/Code/ruby-toy__string-segmentation/lib/dictionary.rb:10: syntax error, unexpected end-of-input, expecting keyword_end (SyntaxError)
rake aborted!
Command failed with status (1): [ruby -I"lib:tests" -I"/Users/mac/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib" "/Users/mac/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rake-10.5.0/lib/rake/rake_test_loader.rb" "tests/*_/__test.rb" ]

File: lib/dictionary.rb [hunk 1]
```diff
@@ -1,7 +1,10 @@
 def valid_word?(word)
   # Populate this array with some words of your choosing.
-  dictionary = []
+  dictionary = ["a","are","you","for","real","yo","cats","go","cat","rain","bow","rainbow","salt","water","saltwater"]
 
-  # Return true if your dictionary contains `word`
+  if dictionary.include?(word)
```
Comment (sumeetjain) [inline lib/dictionary.rb]: You need to `end` a conditional. So

``` ruby
if dictionary.include?(word)
  return true
end
```

Would work.

**Comment by sumeetjain** [comment]
Just so we're clear, your tests are approved.

**Comment by clburns107** [comment]
Ok, i know what i want to say i just need help writing my code to say it. Having trouble iterating through a range.

**Comment by sumeetjain** [comment]
Gotcha. Your code is definitely clear enough to express your intention–the algorithm really does come through. But yes, you're right there are some errors in the code :)

I think the best thing might be for me to provide your code, corrected. I don't usually do that, but yours is so close that it makes the most sense here.

``` ruby
def segment_string(str)
  dictionary= ["cat", "jog"]
  str = "jogcat"

  final = []
  i = 0

  while i <= str.length do
    x = str[0..i]
    if dictionary.include?(x)
      final << x
    end

    i += 1
  end

  return final
end
```

That _should_ be right or at least closer. But let me know if that doesn't help clear up some of the syntax problems/questions you had.

**Comment by clburns107** [comment]
Why is your += operator outside of the range? How does ruby know to use that operator on the range when its outside the loop? 

It also looks like youre not sliceing anything. Would that me more applicable on the harder words?

**Comment by clburns107** [comment]
Also Im confused as to how the dictionary syncs up. apparently some code only works for some people?

**Comment by clburns107** [comment]
I have a loop thats not closed somewhere and i cant see it.

**Comment by clburns107** [comment]
ok i need help with understanding why my code isnt running properly.  I know it has something to do with my pop.length but im not sure if theres something else for what i want it to do

**Pull request closed without merging**